hls-plugin-api/src/Ide/Types.hs

```diff
@@ -162,7 +164,7 @@ defaultConfigDescriptor = ConfigDescriptor True False (mkCustomConfig emptyPrope
 class HasTracing (MessageParams m) => PluginMethod m where
 
   -- | Parse the configuration to check if this plugin is enabled
-  pluginEnabled :: SMethod m -> PluginId -> Config -> Bool
+  pluginEnabled :: SMethod m -> MessageParams m -> PluginDescriptor c -> Config -> Bool
```

Comment:
Is there a bit of an incoherence here? `PluginDescriptor`s can claim to be enabled only on certain file types, but the decision is actually made based on the message params, which need not include a URI. 

Maybe this is fine, we just need to document it. e.g. what happens for methods that _don't_ have a URI in the params? Plugins are on? or off?

Reply:
Indeed, that might be incoherent... I think, the semantics of `pluginEnabled` change with this PR slightly. Previously, it was: "is this plugin disabled from the config?". Now it is "is this plugin disabled or is it even applicable for these parameters?"

Methods that have no URI are not language specific, thus, we only care whether they are disabled.

Since we are conflating two concepts, maybe a completely separate methods makes sense?

Reply:
`canBeHandledByPlugin`? Since this is talking about a _method_ as a thing, and answering the question of whether a particular plugin wants to handle it.

IMO `pluginEnabled` is a misnomer even today, since it's not associated with the plugin itself but with the plugin method handler.

Reply:
> Methods that have no URI are not language specific, thus, we only care whether they are disabled.

This doesn't seem obviously true to me. What about `WorkspaceSymbols`?

Reply:
Indeed, WorkspaceSymbols will probably have to be handled by the function itself. Maybe such a clear cut separation won't be possible in this situation

Reply:
I still think it's valuable, we just need to say what it does and doesn't do. "For plugin methods which include information about which document the request is for, requests which are not associated with a document that has one of the listed file types will not be dispatched to plugin methods for this plugin." or something

Reply:
It is documented now, quite thorough, do you have something like this in mind?